app/domain_models/follow_status_data.rb

```diff
@@ -0,0 +1,83 @@
+class FollowStatusData
+
+    attr_reader :current_status,:current_following,:index
+  
+    def initialize()#將資料庫的紀錄轉換成當下的追蹤狀態
+        @current_status={} #每個使用者的追蹤清單
+        @current_following={}    #每個使用者的粉絲清單
+        @index=1
+        user_number=User.count
+
+        1.upto(user_number) do |user_id|
+            current_status[user_id] = Array.new()
+            current_following[user_id] = Array.new()
+        end
+
+        FollowStatusRecord.all.each do |record|
```

Comment:
Sorry, I do not get your intention here.
Could you easy share your plan about this function?

Please draw a diagram to help me review your code if possible.

Reply:
Like, actually, I think follow and unfollow should be separated.
It could be better to discuss the model before implement.

Reply:
1.initialize
The initialization function of FollowStatusData aims to convert the follow/unfollow record to current follow status(who is following whom and whom is followed by who). 

In the "1.upto(user_number)" section, each user_id(key) in each_user_following_list(hash) and each_user_followed_list(hash) will be assigned with an empty array. The purpose of this section is to initialize the two hashes.

In the "FollowStatusRecord.all.each do" section, each row in follow_status_record table will be loaded and add changes to above-mentions hashes using update_one_record function.

2.update_one_record
This function will convert a row in follow_status_record table(which is one record) into "a change" in current following status.

3.update_record_to_latest
This function will read newly added records from the table to make sure that the current following status that the program is using is in accord with the records in the table.

4.add_follow_status_record
This function will take in subject_user_id, action_type and object_user_id and add the corresponding record to the table. 

memo&reply:
The reason that we did not separate follow and unfollow function here is because these two actions are very similar in terms of the error messages, functions and objects they need. The two function can be easily separated right away but such action will increase the complexity of the project, so we did not choose to do so.